### PR TITLE
LPD 38677 master

### PR DESCRIPTION
**Pull request by titokenzo**
- **LPD-38677 fixes test broken by LPD-23830 (commit a27a9d5268dd36ebe4731b64f517d987b8c95729), the backgoundTask returns an invalid companyId for the test, so a mock for companyId has been added. This also exposes a logical problem when restoring ThreadLocalValues with backgroundTask companyId**
- **LPD-38677 fixes regression from LPD-23830 (commit a27a9d5268dd36ebe4731b64f517d987b8c95729), backgroundTask companyId cannot be used to restore threadLocalValue, as it was used to change the original value**

### CODE REVIEW

**Comment by titokenzo** [comment]
ci:test:relevant

**Comment by titokenzo** [comment]
ci:test:stable

**Comment by titokenzo** [comment]
ci:test:core-functional

File: modules/apps/portal-background-task/portal-background-task-service/src/test/java/com/liferay/portal/background/task/internal/ThreadLocalAwareBackgroundTaskExecutorTest.java [hunk 1]
```diff
@@ -68,6 +69,12 @@ public void testStaleBackgroundTaskIsSkipped() throws Exception {
 					Collections.singletonMap("companyId", 1)))
```
Comment (tinatian) [inline modules/apps/portal-background-task/portal-background-task-service/src/test/java/com/liferay/portal/background/task/internal/ThreadLocalAwareBackgroundTaskExecutorTest.java:69]: @titokenzo We should not need this Collections.singletonMap("companyId", 1)

**Comment by tinatian** [comment]
@titokenzo Please see my comment and resend

**Comment by titokenzo** [comment]
see: https://github.com/liferay-core-infra/liferay-portal/pull/7559

**Pull request closed without merging**